memory/memory_test.go

```diff
@@ -44,53 +53,104 @@ func Test_Storage_Memory_Get(t *testing.T) {
 	result, err := testStore.Get(key)
 	require.NoError(t, err)
 	require.Equal(t, val, result)
+
+	keys, err := testStore.Keys()
+	require.NoError(t, err)
+	require.Len(t, keys, 1)
 }
 
 func Test_Storage_Memory_Set_Expiration(t *testing.T) {
 	var (
-		key = "john"
-		val = []byte("doe")
-		exp = 1 * time.Second
+		testStore = New()
+		key       = "john"
+		val       = []byte("doe")
+		exp       = 1 * time.Second
 	)
 
 	err := testStore.Set(key, val, exp)
 	require.NoError(t, err)
 
 	time.Sleep(1100 * time.Millisecond)
+
+	result, err := testStore.Get(key)
+	require.NoError(t, err)
+	require.Zero(t, len(result))
+
+	keys, err := testStore.Keys()
+	require.NoError(t, err)
+	require.Nil(t, keys)
 }
 
-func Test_Storage_Memory_Get_Expired(t *testing.T) {
-	key := "john"
+func Test_Storage_Memory_Set_Long_Expiration_with_Keys(t *testing.T) {
+	var (
+		testStore = New()
+		key       = "john"
+		val       = []byte("doe")
+		exp       = 5 * time.Second
+	)
+
+	keys, err := testStore.Keys()
+	require.NoError(t, err)
+	require.Nil(t, keys)
 
+	err = testStore.Set(key, val, exp)
+	require.NoError(t, err)
+
+	time.Sleep(1100 * time.Millisecond)
+
+	keys, err = testStore.Keys()
+	require.NoError(t, err)
+	require.Len(t, keys, 1)
+
+	time.Sleep(4000 * time.Millisecond)
 	result, err := testStore.Get(key)
 	require.NoError(t, err)
 	require.Zero(t, len(result))
+
+	keys, err = testStore.Keys()
```

Comment:
i have seen that you always test the new function in other tests 
but would prefer if we had a separate function for the keys functionality so we don't always bloat all the others